clients/da-vinci-client/src/main/java/com/linkedin/davinci/DaVinciBackend.java

```diff
@@ -594,9 +614,13 @@ public void error(String kafkaTopic, int partitionId, String message, Exception
       ingestionReportExecutor.submit(() -> {
         VersionBackend versionBackend = versionByTopicMap.get(kafkaTopic);
         if (versionBackend != null) {
+          /**
+           * Report push status needs to be executed before deleting the {@link VersionBackend}.
+           */
+          reportPushStatus(kafkaTopic, partitionId, ExecutionStatus.ERROR);
+
           versionBackend.completePartitionExceptionally(partitionId, e);
           versionBackend.tryStopHeartbeat();
```

Comment:
Personally, I don't have a preference about heartbeat order.

For error status reporting, it has to go before deleting `VeniceBackend` since `reportPushStatus` will check whether the corresponding `VersionBackend` exists or not:
https://github.com/linkedin/venice/blob/main/clients/da-vinci-client/src/main/java/com/linkedin/davinci/DaVinciBackend.java#L505